### DISCUSSION TITLE
Road to the Godot v4 Asset Lib

### DISCUSSION THREAD

**Discussion by Swivelgames** [lihop/godot-xterm#110] [Ideas]
I was actually in the process of creating an Xterm-powered plugin for Godot when I stumbled on this. I'm a Linux user myself, but I know there's a huge audience out there for Windows users as well.

There's another Godot tool called [YAT - Yet Another Terminal (C#)](https://github.com/MASSHUU12/godot-yat), which looks promising, but it's also in the works and appears to be heavily limited to Windows (the opposite problem this project has).

As a huge proponent of cross-platform, the game I'm working on relies heavily on a vt220-compliant terminal emulator.

I noticed the [Godot v4](https://github.com/lihop/godot-xterm/milestone/1) Milestone. Is there anything else that needs to be knocked out to reach this milestone? It would be fantastic to get an idea of what is left to complete to get `godot-xterm` to the Godot v4 finish line. Maybe with enough insight into what the current issues are, we could get a rough roadmap of what is left to complete, and start hacking away at it!

I'd be interested in helping out where I can!

**Comment by super-Colin** [comment]
I also stumbled upon this; though I was just looking for a simple CLI with some prebuilt functionality, not a full blown Linux terminal emulator! This is amazing!

I'd love to incorporate this into my little productivity/work buddy!!
https://github.com/super-Colin/Desktop-Pet

I love open source, but unfortunately I've not actually contributed code to any open source repos I didn't create... This project and this moment feels right though; how can I help bring this up to Godot 4.4?! (Particularly in the Windows support area?)

**Comment by AlexanderTreml** [comment]
I just opened a pull request regarding Godot v4 and Windows support. If it gets merged, the project might only be a few finishing touches away from being ready for the asset lib.

**Reply by Swivelgames** [reply]
Excited to see this through! :raised_hands: 

For this interested, they can follow this PR:
- https://github.com/lihop/godot-xterm/pull/111

**Reply by lihop** [reply]
Now that #111 has been merged, I've updated the [Godot 4 Asset Library milestone](https://github.com/lihop/godot-xterm/milestone/1).

I'm keen to get the asset up in to the asset library in the next few weeks. I can take care of the two must-do tasks which are #112 and #113. Once #112 is done, I'll create a pre-release with the resulting archive.

#114 is pretty important to resolve, as it results in a crash.
#115 would make the Godot 4 version of the addon have feature parity with the 3.5 version.
#62 isn't a deal breaker, just a bit of ugly UI in an edge-case situation.

Apart from these issues, any testing (particularly on Windows) would be much appreciated.

I know I've been neglecting this addon for a while, so thanks for your patience! And thanks to @AlexanderTreml for implementing windows support which has been a great motivator to finally get this across the line!

**Reply by AlexanderTreml** [reply]
Will start working on #114 next week.

**Comment by lihop** [comment]
Good news! GodotXterm is now available in the Godot asset library: https://godotengine.org/asset-library/asset/4301 for Godot v4.3+.

**Reply by Swivelgames** [reply]
Wooo!! :raised_hands: Thanks @lihop :+1:
